### PR TITLE
Ticket8364 msg log auto truncate

### PR DESCRIPTION
**Pull request by iangillingham-stfc**
https://github.com/ISISComputingGroup/IBEX/issues/8364
EPICS-IocLogServer.git repo modified to facilitate automatic truncation of the 'message' table in 'msg_log' database.
All necessary procedures and scheduled events added to database creation batch files.
Tests available under tests\ directory.

### CODE REVIEW

File: binary-search.sql [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+		SELECT id FROM message ORDER BY id LIMIT 1 INTO first_row;
+        SET p_first_row_id = first_row; -- remember this so it can be returned from the call.
+
+		SELECT concat('binary_search_time(): Get the initial first and last row IDs: first_row: ',first_row,'  last_row: ',last_row) INTO dbg_txt;
+		CALL debug_log(dbg_txt);
+		
+        -- Get the corresponding first and last timestamps
+		SELECT createTime from message where id = last_row limit 1 INTo last_time;
+        SET target_time = ADDTIME(last_time, -p_retention_period);
+
```
[the remark this answers is not in the document]
Comment (iangillingham-stfc) [reply binary-search.sql:63]: The reason for implementing a binary search is that the createTime field is not indexed and would be prohibitively slow when used in a SELECT over a large range of rows. In a binary search, we are requesting the crateTime for a single row of given ID.

[the remark this answers is not in the document]
Comment (Chsudeepta) [reply binary-search.sql:63]: How will it do if we add the index on createTime?

File: truncate_message_table.sql [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+				-- Delete the chuck
+				DELETE FROM message WHERE id >= @a AND id <  @z AND id <target_row;
+				-- Iterate again starting at the next chunk
+				SET @a = @z;
+				DO SLEEP(sleep_period_sec);  -- play nicely!
+			END LOOP chunk;
+			-- Last chunk: We have a chunk with fewer rows than the defined chunk size,
+			-- so just delete the remaining rows up to the defined last row of the overall deletion.
+			DELETE FROM message WHERE id BETWEEN @a AND target_row;
+
```
[the remark this answers is not in the document]
Comment (iangillingham-stfc) [reply truncate_message_table.sql:96]: I think it's clear as it is and works for the way I'm thinking.

File: truncate_message_table.sql [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+			SET @a = first_row_id;
+			chunk: LOOP
+				-- Determine the end id of this chunk to delete
+				SELECT id FROM message WHERE id >= @a ORDER BY id LIMIT chunk_size,1 INTO @z;
+				IF @z IS NULL OR (@z + chunk_size) >= target_row THEN
+					LEAVE chunk;  -- last chunk is less than chunk size
+				END IF;
+				-- Delete the chuck
+				DELETE FROM message WHERE id >= @a AND id <  @z AND id <target_row;
+				-- Iterate again starting at the next chunk
```
[the remark this answers is not in the document]
Comment (iangillingham-stfc) [reply truncate_message_table.sql:92]: Yes, you're correct - well spotted!

**Pull request merged**